test/libsolidity/semanticTests/isoltestTesting/balance_other_contract.sol

```diff
@@ -1,30 +1,9 @@
-contract Other {
+contract C {
     constructor() payable {
-    }
-    function getAddress() public returns (address) {
-        return address(this);
-    }
-}
-contract ClientReceipt {
-    Other other;
-    constructor() payable {
-        other = new Other{value:500}();
-    }
-    function getAddress() public returns (address) {
-        return other.getAddress();
+        payable(address(0x1234)).transfer(500);
     }
 }
 // ----
 // constructor(), 2000 wei ->
-// gas irOptimized: 114353
-// gas irOptimized code: 58800
-// gas legacy: 118617
-// gas legacy code: 111400
-// gas legacyOptimized: 114067
-// gas legacyOptimized code: 59800
 // balance -> 1500
-// gas irOptimized: 191881
-// gas legacy: 235167
-// gas legacyOptimized: 180756
-// getAddress() -> 0x137aa4dfc0911524504fcd4d98501f179bc13b4a
-// balance: 0x137aa4dfc0911524504fcd4d98501f179bc13b4a -> 500
+// balance: 0x0000000000000000000000000000000000001234 -> 500
```

Comment:
Here the problem was that the `balance` helper requires hard-coding the address and it will be different on EOF.

To test balance, we don't really need to deploy a new contract though. We can just send some value to a randomly chosen address and then it won't change on EOF.